Src/ST-LIB_LOW/StateMachine/StateMachine.cpp

```diff
@@ -105,6 +109,13 @@ StateMachine::StateMachine(uint8_t initial_state) :
 	initial_state(initial_state), current_state(initial_state) {
 	add_state(initial_state);
 	enter_state(initial_state);
+
+	#ifdef SIM_ON
+		SharedMemory::start_emulated_state_machine();
+		uint8_t counter=SharedMemory::state_machine_sm[0]+1;
```

Comment:
```suggestion
		uint8_t counter = SharedMemory::state_machine_sm[0]++;
```

Reply:
probably just get rid of the `update_state_machine_counter()`, and simply do `SharedMemory::state_machine_sm[0]++`  because this suggestion you just did does exactly the opposite of what you want it to do, take a look.

Reply:
noup, in any case you should pre-increment